mmpose/models/losses/mesh_loss.py

```diff
@@ -0,0 +1,337 @@
+import torch
+import torch.nn as nn
+
+from ..mesh_heads.geometric_layers import batch_rodrigues
+from ..registry import LOSSES
+
+
+def perspective_projection(points, rotation, translation, focal_length,
+                           camera_center):
+    """This function computes the perspective projection of a set of 3D points.
+
+    Notes:
+        batch size: B
+        point number: N
+
+    Args:
+        points (Tensor([B, N, 3])): A set of 3D points
+        rotation (Tensor([B, 3, 3])): Camera rotation matrix
+        translation (Tensor([B, 3])): Camera translation
+        focal_length (Tensor([B,])): Focal length
+        camera_center (Tensor([B, 2])): Camera center
+
+    Returns:
+        projected_points (Tensor([B, N, 3])): Projected 2D
+            points in image space.
+    """
+
+    batch_size = points.shape[0]
+    K = torch.zeros([batch_size, 3, 3], device=points.device)
```

Comment:
since K is the same for all samples in a batch, maybe try this

```python
K = torch.zeros([3, 3], device=points.device)
...
projected_points = torch.einsum('ij,bkj->bki', K, projected_points)
```

Reply:
K can be different for samples in a batch, if the focal_length and camera_center of samples are different.
So I prefer to keep K in shape [batch_size, 3, 3].

Reply:
yeah you are right